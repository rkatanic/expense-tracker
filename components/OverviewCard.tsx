interface Props {
  icon: JSX.Element;
  primaryValue: number;
  secondaryValue: number;
  primaryDescription: string;
  secondaryDescription: string;
}

const OverviewCard = ({
  icon,
  primaryValue,
  secondaryValue,
  primaryDescription,
  secondaryDescription,
}: Props): JSX.Element => {
  return (
    <div className="overview-card">
      <div className="overview-card-header">
        <div className="overview-card-icon">{icon}</div>
        <div className="overview-card-primary-text-wrapper">
          <div className="overview-card-primary-value">
            {primaryValue} <span className="overview-card-currency">BAM</span>
          </div>
          <div className="overview-card-primary-description">
            {primaryDescription}
          </div>
        </div>
      </div>
      <div className="overview-card-footer">
        <div className="overview-card-secondary-description">
          {secondaryDescription}
        </div>
        <div className="overview-card-secondary-value">{secondaryValue}</div>
      </div>
    </div>
  );
};

export default OverviewCard;
