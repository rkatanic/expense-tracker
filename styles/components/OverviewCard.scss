.overview-card {
    padding: 1.5rem;
    background-color: var(--color-primary-300);
    width: 16rem;
    height: 16rem;
    border-radius: 8px;
    color: var(--color-black-one);
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    &-currency {
        font-size: 0.875rem;
    }

    &-primary {
        &-value {
            font-size: 1.5rem;
            font-weight: 600;
        }
    }

    &-secondary {
        &-value {
            border: 1px solid var(--color-black-one);
            border-radius: 10rem;
            width: 3rem;
            height: 3rem;
            display: flex;
            justify-content: center;
            align-items: center;
            font-weight: 00;
            font-size: 0.875rem;
        }

        &-description {
            font-weight: 500;
        }
    }

    &-header,
    &-footer {
        display: flex;
        align-items: center;
        justify-content: space-between;
    }

    &-header {
        gap: 1.5rem;
        justify-content: flex-start;
    }

    &-footer {
        align-items: flex-end;
    }

    &:nth-child(2) {
        background-color: var(--color-primary-600);
    }


    &:nth-child(3) {
        background-color: var(--color-primary-900);
    }
}