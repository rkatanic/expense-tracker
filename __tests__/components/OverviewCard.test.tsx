import { render } from "@testing-library/react";
import OverviewCard from "../../components/OverviewCard";

describe("OverviewCard", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <OverviewCard
        icon={<>icon</>}
        primaryValue={100}
        secondaryValue={1}
        primaryDescription={"primary text"}
        secondaryDescription={"secondary text"}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
