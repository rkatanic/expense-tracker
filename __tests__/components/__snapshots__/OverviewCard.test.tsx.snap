// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewCard should render 1`] = `
<body>
  <div>
    <div
      class="overview-card"
    >
      <div
        class="overview-card-header"
      >
        <div
          class="overview-card-icon"
        >
          icon
        </div>
        <div
          class="overview-card-primary-text-wrapper"
        >
          <div
            class="overview-card-primary-value"
          >
            100
             
            <span
              class="overview-card-currency"
            >
              BAM
            </span>
          </div>
          <div
            class="overview-card-primary-description"
          >
            primary text
          </div>
        </div>
      </div>
      <div
        class="overview-card-footer"
      >
        <div
          class="overview-card-secondary-description"
        >
          secondary text
        </div>
        <div
          class="overview-card-secondary-value"
        >
          1
        </div>
      </div>
    </div>
  </div>
</body>
`;
